types/accesscontrol/access_operation_map.go

```diff
@@ -7,22 +7,23 @@ type MessageAccessOpsChannelMapping = map[int]AccessOpsChannelMapping
 type AccessOpsChannelMapping = map[AccessOperation][]chan interface{}
 
 func WaitForAllSignalsForTx(messageIndexToAccessOpsChannelMapping MessageAccessOpsChannelMapping) {
-	for _, accessOpsToChannelsMap  := range messageIndexToAccessOpsChannelMapping {
+	for _, accessOpsToChannelsMap := range messageIndexToAccessOpsChannelMapping {
 		for _, channels := range accessOpsToChannelsMap {
 			for _, channel := range channels {
 				<-channel
+				// reenqueue so that other reads can also be unblocked
+				channel <- struct{}{}
```

Comment:
why do we need to reenqueue? each dag edge has its own channel so we never need to write to a channel twice.